app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,33 @@
+class CommentsController < ApplicationController
+  before_action :get_parent
+
+  def create
+    @comment = @parent.comments.new(comment_params.merge(user: current_user))
+    
+    if @comment.save
+      flash[:notice] = "Comment saved."
+      redirect_to article_path(@comment.parent_post)
+    else
+      flash[:alert] = "Unable to save comment. Please try again."
+      #If there is a better way to write this logic, please tell me. 
+      #Apparently, inline if-else doesn't syntactically work
+      if @parent.is_a?(Article)
+        redirect_to article_path(@parent)
+      else
+        redirect_to article_path(@parent.parent_post)
+      end
+    end
+  end
+
+  private
+  def get_parent
+    @parent = Article.find(params[:article_id]) if params[:article_id]
+    @parent = Comment.find(params[:comment_id]) if params[:comment_id]
```

Comment:
honestly, I'm a little surprised that `@parent` isn't set to `nil` here if `params[:comment_id]` is `nil`, but I think what's here is correct.

Another approach could look either like:

```ruby
resource, id = request.path.split('/')[1, 2]
@commentable = resource.singularize.classify.constantize.find(id)
```

or

```ruby
klass = [Article, Comment].detect { |c| params["#{c.name.underscore}_id"] }
@commentable = klass.find(params["#{klass.name.underscore}_id"])
```